core/templates/filters/string-utility-filters/normalize-whitespace-punctuation-and-case.pipe.ts

```diff
@@ -29,40 +29,36 @@ import { Injectable, Pipe, PipeTransform } from '@angular/core';
 export class NormalizeWhitespacePunctuationAndCasePipe
 implements PipeTransform {
   transform(input: string): string {
-    if (typeof input === 'string') {
-      let isAlphanumeric = function(character: string) {
-        return 'qwertyuiopasdfghjklzxcvbnm0123456789'.indexOf(
-          character.toLowerCase()) !== -1;
-      };
+    let isAlphanumeric = function(character: string) {
+      return 'qwertyuiopasdfghjklzxcvbnm0123456789'.indexOf(
+        character.toLowerCase()) !== -1;
+    };
 
-      input = input.trim();
-      let inputLines = input.split('\n');
-      let resultLines = [];
-      for (let i: number = 0; i < inputLines.length; i++) {
-        let result = '';
+    input = input.trim();
+    let inputLines = input.split('\n');
+    let resultLines = [];
+    for (let i: number = 0; i < inputLines.length; i++) {
+      let result = '';
 
-        let inputLine = inputLines[i].trim().replace(/\s{2,}/g, ' ');
-        for (let j: number = 0; j < inputLine.length; j++) {
-          let currentChar: string = inputLine.charAt(j).toLowerCase();
-          if (currentChar === ' ') {
-            if (j > 0 && j < inputLine.length - 1 &&
-                                isAlphanumeric(inputLine.charAt(j - 1)) &&
-                                isAlphanumeric(inputLine.charAt(j + 1))) {
-              result += currentChar;
-            }
-          } else {
+      let inputLine = inputLines[i].trim().replace(/\s{2,}/g, ' ');
+      for (let j: number = 0; j < inputLine.length; j++) {
+        let currentChar: string = inputLine.charAt(j).toLowerCase();
+        if (currentChar === ' ') {
+          if (j > 0 && j < inputLine.length - 1 &&
+                              isAlphanumeric(inputLine.charAt(j - 1)) &&
+                              isAlphanumeric(inputLine.charAt(j + 1))) {
```

Comment:
This doesn't really make sense, so if I do "hello, world" it'll become "hello,world"? That looks worse than the original. I believe you only refactored the original code in this PR, so this is more of a question of what this pipe is actually supposed to do.

Reply:
Ya, basically any spaces that doesn't separate two alphanumeric words will be removed. This is the original behavior of this pipe.

Reply:
Can we file an issue for this @Manan-Rathi. +cc @nithusha21.

Reply:
> Can we file an issue for this @Manan-Rathi. +cc @nithusha21.

Done, filed [#14840](https://github.com/oppia/oppia/issues/14840)!